web/Makefile

```diff
@@ -32,4 +45,90 @@ clean_venv_dev:
 build_dir:
 	mkdir -p $(BUILD_DIR)
 
-# TODO: add targets to build server separately.
+package_dir_structure:
+	mkdir -p $(BUILD_DIR) && \
+	mkdir -p $(CC_BUILD_DIR)/bin && \
+	mkdir -p $(CC_BUILD_LIB_DIR)
+
+package_thrift: thrift package_dir_structure
+	mkdir -p $(CC_BUILD_API_DIR) && \
+	mkdir -p $(CC_BUILD_GEN_DIR) && \
+	cp -r $(BUILD_DIR)/thrift/v*/gen-py/* $(CC_BUILD_GEN_DIR) && \
+	cp -r $(BUILD_DIR)/thrift/v*/gen-js/* $(CC_BUILD_API_DIR)
+
+gen_docs: build_dir
+	cd $(ROOT) && \
+	doxygen $(ROOT)/Doxyfile.in && \
+	cp -a $(ROOT)/gen-docs $(BUILD_DIR)/gen-docs
+
+package_docs: gen_docs
+	mkdir -p $(CC_BUILD_DOCS_DIR) && \
+	mkdir -p $(CC_BUILD_DOCS_DIR)/checker_md_docs && \
+	cp -r $(BUILD_DIR)/gen-docs/html/* $(CC_BUILD_DOCS_DIR) && \
+	cp -r $(ROOT)/docs/web/checker_docs/* $(CC_BUILD_DOCS_DIR)/checker_md_docs/
+
+	# Rename gen-docs to doc.
+	rm -rf $(CC_BUILD_WEB_DIR)/userguide/doc && \
+	mv $(CC_BUILD_WEB_DIR)/userguide/gen-docs $(CC_BUILD_WEB_DIR)/userguide/doc
+
+gen_userguide:
+	mkdir -p $(CC_BUILD_WEB_DIR) && \
+	$(MAKE) -C $(CC_SERVER)/www/userguide
+
+package_www: package_dir_structure
+	cp -r $(CC_SERVER)/www/* $(CC_BUILD_WEB_DIR)
+
+package_userguide: gen_userguide package_www
+	mkdir -p $(CC_BUILD_WEB_DIR)/images && \
+	mv $(CC_BUILD_WEB_DIR)/userguide/images/* $(CC_BUILD_WEB_DIR)/images && \
+	rm -rf $(CC_BUILD_WEB_DIR)/userguide/images
+
+package_vendor: package_dir_structure
+	$(MAKE) -C $(CC_SERVER)/vendor build BUILD_DIR=$(CC_BUILD_WEB_PLUGINS_DIR)
+
+package_web: package_thrift package_userguide package_vendor package_docs
```

Comment:
Please add a comment that this target should be used from the top level `Makefile` to build the package together with the analyzer part.